v2/uuid/pgxcompat/uuid.go

```diff
@@ -0,0 +1,258 @@
+package pgxcompat
+
+import (
+	"database/sql/driver"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgtype"
+
+	"github.com/SKF/go-utility/v2/uuid"
+)
+
+const uuidBinaryLength = 16
+const uuidStrShortLength = 32
+const uuidStrFullLength = 36
+
+var errUndefined = errors.New("cannot encode status undefined")
+var errBadStatus = errors.New("invalid status")
+
+type UUID struct {
+	UUID   uuid.UUID
+	Status pgtype.Status
+}
+
+type Getter interface {
+	Get() interface{}
+}
+
+func (u *UUID) Set(value interface{}) error { // nolint:gocyclo
+	if value == nil {
+		*u = UUID{Status: pgtype.Null}
+		return nil
+	}
+
+	if getterValue, ok := value.(Getter); ok {
```

Comment:
This thing is really difficult to understand

Reply:
I agree. It is for compatibility with nested `pgtype` types. It recursively calls `Get()` on src until it returns itself, and sets dst to the innermost value. Copied from [here](https://github.com/jackc/pgtype/blob/90af821478c74fd8f917b7301d79886f933e5fe3/uuid.go#L21), and I want to be as compatible as possible with the other pgtype types.

Reply:
Something like this would make it a bit more readable for me. 
```
	if getterInterface, isGetter := value.(Getter); isGetter {
		getterValue := getterInterface.Get()
		if getterValue != getterInterface {
			return u.Set(getterValue)
		}
	}
```

This thing i found strange:
`if getterValue != getterInterface`
 but perhaps thats just my golang experience talking.
Does it mean `the interface returned is oft he same type as Getter` ?

Reply:
I guess it's the getter just returning itself? I don't know the details about why this is here, it's just boilerplate when implementing a custom type for the pgx driver.